# Profitable App Profiles for the Apple Store and Google Play Markets

In this project an analysis on Apple Store and Google Play free-only applications will be made. The objective is to understand which apps are more profitable utilising in-app ads. Monitoring the interactions users have with each app is the better way to understand which ones stand out giving developers a direction to head regarding what kind of apps to make. 

## Opening and exploring the data
In order to be a little less time consuming, I'll be opening a data set with information about ~10000 Google Play apps and ~7000 Apple Store apps, instead of collecting the information about all the approximatelly 4 million combined apps in both stores.

In [2]:
from csv import reader

'''opening the google play data set'''
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
googleapps = list(read_file)
google_apps_data = googleapps[1:]

'''opening the App Store data set'''
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
appleapps = list(read_file)
apple_apps_data = appleapps[1:]



In [7]:
'''
Data exploration function - printing the rows for the chosen dataset, from star row to end row with a blank 
line spacing each row
'''
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
print(googleapps[0])
print('\n')
gplay  = explore_data(google_apps_data,0,4, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


For the google play data set apps we can see that there is data from 10841 apps and 13 categories (columns). The categories that at first can be useful are: App, Category, Rating, Reviews, Installs, Price and Genres. 

If there is any questions regarding any category, here is a link to the [data set](https://www.kaggle.com/lava18/google-play-store-apps) documentation, where more information is available. 

Next is the Apple Store exploration. 

In [8]:
print(appleapps[0])
print('\n')
appstore = explore_data(apple_apps_data,0,4,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns: 16


Regarding the Apple store data set, there are 7197 apps and 16 categories. The ones that can be interesting are: track_name, currency, price, rating_count_tot, rating_count_ver and prime_genre. For some help on categories that may be not so clear, here is a [link to documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) on the data set.